packages/react-router/index.js

```diff
@@ -522,14 +543,11 @@ export function useRoutes(routes, basename = '', caseSensitive = false) {
   // Otherwise render an element.
   let element = matches.reduceRight((outlet, { params, pathname, route }) => {
     return (
-      <RouteContext.Provider
-        children={route.element}
-        value={{
-          outlet,
-          params: readOnly({ ...parentParams, ...params }),
-          pathname: joinPaths([basename, pathname]),
-          route
-        }}
+      <RouterContextWrapper
```

Comment:
Use memo in loops is forbidden. This is the reason of bringing in `RouterContextWrapper `